app/scripts/services/blockTagTypeService.ts

```diff
@@ -0,0 +1,113 @@
+import angular from 'angular';
+import { $Http, $Q } from 'injectables';
+
+export interface BlockTagType {
+  id: string | null;
+  ministryId: string | null;
+  name: string;
+  prettyName: string;
+}
+
+export interface BlockTagTypeMapping {
+  blockId: string;
+  title: string;
+  blockTagTypeId: string | null;
+}
+
+export interface ValidationResult {
+  valid: boolean;
+  message: string;
+}
+
+export class BlockTagTypeService {
+  private readonly $http: $Http;
+  private readonly $q: $Q;
+  private readonly defaultBlockTagType: BlockTagType = {
+    id: null,
+    ministryId: null,
+    name: 'None',
+    prettyName: 'None',
+  };
+  private cachedBlockTagTypes: BlockTagType[] = [this.defaultBlockTagType];
+  private previousConferenceId: string | null = null;
+
+  constructor($http: $Http, $q: $Q) {
+    this.$http = $http;
+    this.$q = $q;
+  }
```

Comment:
TypeScript has a shortcut for this.

```suggestion
  constructor(private $http: $Http, private $q: $Q) {}
```